inst/update_deployer.R

```diff
@@ -0,0 +1,41 @@
+#' Update a package in a deployer from the source code
+#'
+#' @param pkg the path to a package source
+#' @param repo path to a drat repo. Defaults to both the current and 'drat' repo 
+#' @param internet a future parameter to update packages from internet if it is
+#'   available.
+#' @examples
+#'
+#' deploy_package("../linelist") # update the linelist package, assuming its
+#'   source lives one directory upstream of this deployer
+deploy_package <- function(pkg = NULL, repo = c(here::here(), here::here("drat")), internet = FALSE) {
+  if (is.null(pkg)) {
+    msg <- paste(
+      "Please supply a path to a package source"
+    )
+    stop(msg)
+  }
+  stopifnot(file.exists(pkg))
+  
+  vers <- read.dcf(file.path(pkg, "DESCRIPTION"))[, "Version"]
+  message("Building source package ...")
+  src <- devtools::build(pkg, path = here::here(), binary = FALSE)
```

Comment:
Is `here` or `devtools` guaranteed to be installed on the host machine - neither are dependencies of `nomad`. I would suggest avoiding `devtools::build` here unless you want it to go and upgrade everything in an unpredictable, unreliable and likely unwelcome way (it will also behave differently to `nomad` wrt `Remotes:` - nomad ignores these because they're a great idea until you end up with `devtools` upgrading things like Rcpp and breaking absolutely everything)

Reply:
> Is `here` or `devtools` guaranteed to be installed on the host machine - neither are dependencies of `nomad`. 

Good point. These are both on the deployers, so I guess this may be deployer-specific for now

> I would suggest avoiding `devtools::build` here unless you want it to go and upgrade everything in an unpredictable, unreliable and likely unwelcome way (it will also behave differently to `nomad` wrt `Remotes:` - nomad ignores these because they're a great idea until you end up with `devtools` upgrading things like Rcpp and breaking absolutely everything)

devtools stopped forcing upgrades last October.